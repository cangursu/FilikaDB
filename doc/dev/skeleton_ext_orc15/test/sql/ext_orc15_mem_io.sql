-- simple input
SELECT '120'::big_ext_orc15;
SELECT '3c'::big_ext_orc15;
-- case insensitivity
SELECT '3C'::big_ext_orc15;
SELECT 'FoO'::big_ext_orc15;
-- invalid characters
SELECT 'foo bar'::big_ext_orc15;
SELECT 'abc$%2'::big_ext_orc15;
-- negative values
SELECT '-10'::big_ext_orc15;
-- to big values
SELECT 'abcdefghijklmn'::big_ext_orc15;

-- storage
BEGIN;
CREATE TABLE ext_orc15_test(val big_ext_orc15);
INSERT INTO ext_orc15_test VALUES ('123'), ('3c'), ('5A'), ('zZz');
SELECT * FROM ext_orc15_test;
UPDATE ext_orc15_test SET val = '567a' where val = '123';
SELECT * FROM ext_orc15_test;
UPDATE ext_orc15_test SET val = '-aa' where val = '3c';
SELECT * FROM ext_orc15_test;
ROLLBACK;
