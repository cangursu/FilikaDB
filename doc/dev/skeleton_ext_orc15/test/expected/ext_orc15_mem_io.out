-- simple input
SELECT '120'::big_ext_orc15;
 big_ext_orc15 
---------------
 120
(1 row)

SELECT '3c'::big_ext_orc15;
 big_ext_orc15 
---------------
 3c
(1 row)

-- case insensitivity
SELECT '3C'::big_ext_orc15;
 big_ext_orc15 
---------------
 3c
(1 row)

SELECT 'FoO'::big_ext_orc15;
 big_ext_orc15 
---------------
 foo
(1 row)

-- invalid characters
SELECT 'foo bar'::big_ext_orc15;
ERROR:  invalid input syntax for big_ext_orc15: "foo bar"
LINE 1: SELECT 'foo bar'::big_ext_orc15;
               ^
SELECT 'abc$%2'::big_ext_orc15;
ERROR:  invalid input syntax for big_ext_orc15: "abc$%2"
LINE 1: SELECT 'abc$%2'::big_ext_orc15;
               ^
-- negative values
SELECT '-10'::big_ext_orc15;
 big_ext_orc15 
---------------
 -10
(1 row)

-- to big values
SELECT 'abcdefghijklmn'::big_ext_orc15;
ERROR:  value "abcdefghijklmn" is out of range for type big_ext_orc15
LINE 1: SELECT 'abcdefghijklmn'::big_ext_orc15;
               ^
-- storage
BEGIN;
CREATE TABLE ext_orc15_test(val big_ext_orc15);
INSERT INTO ext_orc15_test VALUES ('123'), ('3c'), ('5A'), ('zZz');
SELECT * FROM ext_orc15_test;
 val 
-----
 123
 3c
 5a
 zzz
(4 rows)

UPDATE ext_orc15_test SET val = '567a' where val = '123';
SELECT * FROM ext_orc15_test;
 val  
------
 3c
 5a
 zzz
 567a
(4 rows)

UPDATE ext_orc15_test SET val = '-aa' where val = '3c';
SELECT * FROM ext_orc15_test;
 val  
------
 5a
 zzz
 567a
 -aa
(4 rows)

ROLLBACK;
